### PR TITLE
feat(std/io): add LimitedReader

### PR DESCRIPTION
**Pull request by hastri**
#6019

### CODE REVIEW

File: std/io/readers.ts [hunk 1]
```diff
@@ -1,6 +1,7 @@
 // Copyright 2018-2020 the Deno authors. All rights reserved. MIT license.
 type Reader = Deno.Reader;
 import { encode } from "../encoding/utf8.ts";
+export * from "./limited_reader.ts";
```
Comment (ry) [inline std/io/readers.ts:4]: probably better to just move LimitedReader into this module

Comment (hastri) [reply std/io/readers.ts:4]: I wanted to, but was not sure about the reference to the go module / license notice

Comment (ry) [reply std/io/readers.ts:4]: You can just put it at the top of the file... I think everything in here is ported from go

File: std/io/readers_test.ts [hunk 2]
```diff
@@ -36,3 +37,15 @@ test("ioMultiReader", async function (): Promise<void> {
...
+  let r = new LimitedReader(new StringReader("abc"), 2);
+  let buffer = await Deno.readAll(r);
+  assertEquals(decode(buffer), "ab");
+  r = new LimitedReader(new StringReader("abc"), 3);
+  buffer = await Deno.readAll(r);
+  assertEquals(decode(buffer), "abc");
+  r = new LimitedReader(new StringReader("abc"), 4);
+  buffer = await Deno.readAll(r);
+  assertEquals(decode(buffer), "abc");
+});
```
Comment (ry) [inline std/io/readers_test.ts:51]: Consider adding this test as well: https://github.com/golang/go/blob/f8662a5a96698c5fcf3a77f692e425c7b857e681/src/io/io_test.go#L35-L48

Comment (hastri) [reply std/io/readers_test.ts:63]: Added

**Pull request merged**